### PR TITLE
AlterJobV2 Pending Status Replay Fix Bug

### PR DESCRIPTION
**Pull request by wangbo**
#2920

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 1]
```diff
@@ -1702,8 +1702,18 @@ public long loadAlterJob(DataInputStream dis, long checksum, JobType type) throw
             newChecksum ^= size;
             for (int i = 0; i < size; i++) {
                 AlterJobV2 alterJobV2 = AlterJobV2.read(dis);
-                if (type == JobType.ROLLUP) {
-                    this.getRollupHandler().addAlterJobV2(alterJobV2);
+                if (type == JobType.ROLLUP || type == JobType.SCHEMA_CHANGE) {
+                    if (type == JobType.ROLLUP)
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/catalog/Catalog.java]: code style. Add {} to if else.

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 2]
```diff
@@ -1702,8 +1702,18 @@ public long loadAlterJob(DataInputStream dis, long checksum, JobType type) throw
...
+                        this.getRollupHandler().addAlterJobV2(alterJobV2);
+                    else
+                        alterJobsV2.put(alterJobV2.getJobId(), alterJobV2);
+                    // ATTN : we just want to add tablet into TabletInvertedIndex when only PendingJob is checkpointed
+                    // to prevent TabletInvertedIndex data loss,
+                    // So just use AlterJob.replay() instead of AlterHandler.replay().
+                    if (alterJobV2.getJobState() == AlterJobV2.JobState.PENDING) {
+                        alterJobV2.replay(alterJobV2);
+                        LOG.warn("replay pending alter job when load alter job {} ", alterJobV2.getJobId());
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/catalog/Catalog.java]: info is enough.

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java (hunk 1 above)
Comment (wangbo) [reply fe/src/main/java/org/apache/doris/catalog/Catalog.java]: 👌

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java (hunk 2 above)
Comment (wangbo) [reply fe/src/main/java/org/apache/doris/catalog/Catalog.java]: 👌

**Review by kangkaisen — COMMENTED** [review]
+1, LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**